ndc-test/src/error.rs

```diff
@@ -65,6 +65,10 @@ pub enum Error {
     BenchmarkExceededTolerance(f64),
     #[error("response from connector does not satisfy requirement: {0}")]
     ResponseDoesNotSatisfy(String),
+    #[error("invalid response at path {}: expected {1}", .0.join("."))]
```

Comment:
`.0.join(".")` this is interesting. I am assuming `0` is a string and then you join with a `.`, but why the extra `.` at the start?

Reply:
`.0` is just `#[error]` syntax for "the first argument", so this is "take the first argument (an array) and join it with `.`".